modules/programs/fish.nix

```diff
@@ -99,33 +124,69 @@ in
 
   config = mkIf cfg.enable {
 
-    environment.etc."fish/foreign-env/shellInit".text = cfge.shellInit;
-    environment.etc."fish/foreign-env/loginShellInit".text = cfge.loginShellInit;
-    environment.etc."fish/foreign-env/interactiveShellInit".text = cfge.interactiveShellInit;
-
-    environment.etc."fish/nixos-env-preinit.fish".text = ''
+    environment = mkMerge [
+      (mkIf cfg.translateEnvironment
+      {
+        etc."fish/set-environment.fish".source = pkgs.runCommand "set-environment.fish" {
+        nativeBuildInputs = [ babelfish ];
+      } ''
+        ${babelfish}/bin/babelfish < ${config.system.build.setEnvironment} > $out;
+      '';
+
+       etc."fish/shellInit".source = pkgs.runCommand "shellInit" {
```

Comment:
fish files always have .fish at the end, so I think you should add that here too.